### PR TITLE
PWGHF: Final version Lb tasks

### PR DESCRIPTION
**Pull request by mjongerh**
Final version of Lb tasks

O2Physics: https://github.com/AliceO2Group/O2Physics/pull/994

### CODE REVIEW

File: codeHF/config_tasks.sh [hunk 1]
```diff
@@ -260,6 +271,8 @@ function MakeScriptO2 {
   [ "$ISALICE3" -eq 1 ] && SUFFIX_ALICE3="-alice3" || SUFFIX_ALICE3=""
 
   WORKFLOWS=""
+  # MC converter to version 001 of mcparticles
+  [ $DOO2_MCCONV -eq 1 ] && WORKFLOWS+=" o2-analysis-mc-converter"
```
Comment (vkucera) [inline codeHF/config_tasks.sh:275]: Remove (already there)

**Review by vkucera — CHANGES_REQUESTED** [review]

File: codeHF/workflows.yml [hunk 2]
```diff
@@ -108,6 +119,11 @@ workflows:
       - o2-analysis-pid-tof-full
     tables: HFSELDPLUSCAND
 
+  o2-analysis-hf-lb-tolcpi-candidate-selector:
+    <<: *selector_3prong
```
Comment (vkucera) [inline codeHF/workflows.yml:123]:
```suggestion
```
There is nothing that could be used from that inheritance

File: codeHF/workflows.yml [hunk 3]
```diff
@@ -83,6 +83,17 @@ workflows:
       default: [HFCANDBPLUSBASE, HFCANDBPLUSEXT]
       mc: [HFCANDBPMCREC, HFCANDBPMCGEN]
 
+  o2-analysis-hf-candidate-creator-lb:
+    <<: *cand_creator
+    options:
+      mc: "--doMC"
```
Comment (vkucera) [inline codeHF/workflows.yml:89]: This should be removed together with replacing the `--doMC` option by a `processMC` switch ASAP.

File: codeHF/workflows.yml [hunk 4]
```diff
@@ -156,6 +172,11 @@ workflows:
   o2-analysis-hf-task-lc:
     dependencies: o2-analysis-hf-lc-candidate-selector
 
+  o2-analysis-hf-task-lb:
+    options:
+      mc: "--doMC"
```
Comment (vkucera) [inline codeHF/workflows.yml:177]: This should be removed together with replacing the `--doMC` option by a `processMC` switch ASAP.

File: codeHF/dpl-config_run5_hf.json [hunk 5]
```diff
@@ -643,6 +687,28 @@
         },
        "processMC": "true"
     },
+    "hf-task-lb": {
+        "cutYCandMax": "2.0",
+        "d_selectionFlagLb": "0",
```
Comment (vkucera) [inline codeHF/dpl-config_run5_hf.json:692]:
```suggestion
        "selectionFlagLb": "0",
```

File: codeHF/dpl-config_run5_hf.json [hunk 6]
```diff
@@ -643,6 +687,28 @@
...
+                "5",
+                "7",
+                "10",
+                "13",
+                "16",
+                "20",
+                "24"
+            ]
+        },
+       "processMC": "true"
```
Comment (vkucera) [inline codeHF/dpl-config_run5_hf.json:710]:
```suggestion
        }
```
There is no such switch.

File: codeHF/config_tasks.sh [hunk 7]
```diff
@@ -45,6 +45,7 @@ DOO2_CAND_2PRONG=0  # hf-candidate-creator-2prong
 DOO2_CAND_3PRONG=0  # hf-candidate-creator-3prong
 DOO2_CAND_CASC=0    # hf-candidate-creator-cascade
 DOO2_CAND_X=0       # hf-candidate-creator-x
+DOO2_CAND_LB=0      # hf-candidate-creator-lb
```
Comment (vkucera) [inline codeHF/config_tasks.sh:48]:
```suggestion
DOO2_CAND_LB=0      # hf-candidate-creator-lb
DOO2_CAND_X=0       # hf-candidate-creator-x
```
Lb is before X everywhere else.

File: codeHF/config_tasks.sh [hunk 8]
```diff
@@ -278,6 +289,7 @@ function MakeScriptO2 {
   [ $DOO2_CAND_2PRONG -eq 1 ] && WORKFLOWS+=" o2-analysis-hf-candidate-creator-2prong"
   [ $DOO2_CAND_3PRONG -eq 1 ] && WORKFLOWS+=" o2-analysis-hf-candidate-creator-3prong"
   [ $DOO2_CAND_X -eq 1 ] && WORKFLOWS+=" o2-analysis-hf-candidate-creator-x"
+  [ $DOO2_CAND_LB -eq 1 ] && WORKFLOWS+=" o2-analysis-hf-candidate-creator-lb"
```
Comment (vkucera) [inline codeHF/config_tasks.sh:292]:
```suggestion
  [ $DOO2_CAND_LB -eq 1 ] && WORKFLOWS+=" o2-analysis-hf-candidate-creator-lb"
  [ $DOO2_CAND_X -eq 1 ] && WORKFLOWS+=" o2-analysis-hf-candidate-creator-x"
```

File: codeHF/config_tasks.sh [hunk 9]
```diff
@@ -392,6 +407,7 @@ function MakeScriptPostprocess {
     [ $DOO2_TASK_D0 -eq 1 ] && { OPT_COMPARE+=" d0 "; [ "$ISMC" -eq 1 ] && OPT_COMPARE+=" d0-mc "; }
     [ $DOO2_TASK_DPLUS -eq 1 ] && OPT_COMPARE+=" dplus "
     [ $DOO2_TASK_LC -eq 1 ] && { OPT_COMPARE+=" lc "; [ "$ISMC" -eq 1 ] && OPT_COMPARE+=" lc-mc "; }
+    [ $DOO2_TASK_LB -eq 1 ] && { OPT_COMPARE+=" lb "; [ "$ISMC" -eq 1 ] && OPT_COMPARE+=" lb-mc "; }
```
Comment (vkucera) [inline codeHF/config_tasks.sh:410]:
```suggestion
```
There is no comparison implemented for Lb in `Compare.C`.

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
Hi @mjongerh , any updates on this?

**Comment by vkucera** [comment]
@mjongerh did you manage to progress with this?

File: codeHF/dpl-config_run5_hf.json (hunk 5 above)
Comment (mjongerh) [reply codeHF/dpl-config_run5_hf.json:819]: Is there any particular reason why some use the d_ prefix and some don't?

Comment (vkucera) [reply codeHF/dpl-config_run5_hf.json:819]: The first tasks used the prefix and some other tasks took it from there. It has been fixed somewhere but not everywhere yet. The conventional way is without the prefix.

File: codeHF/workflows.yml (hunk 2 above)
Comment (mjongerh) [reply codeHF/workflows.yml:120]: Has been removed

**Review by vkucera — APPROVED** [review]

**Pull request merged**